File: src/main/java/com/wishai/xzrtw/model/block/ImageBlock.java
package com.wishai.xzrtw.model.block;


import com.wishai.xzrtw.model.File;
import com.wishai.xzrtw.model.gui.blocks.BlockEle;
import com.wishai.xzrtw.model.gui.blocks.ImageBlockEle;
import com.wishai.xzrtw.service.ResourceService;

import javax.persistence.*;

@Entity
@Table(name = "block_image")
public class ImageBlock extends Block {

    private String info;

    /* link to file */
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "fileId")
    private File file;

    public ImageBlock() {
        super.setdType(ResourceService.BLOCK_TYPE_IMAGE);
    }

    public String getInfo() {
        return info;
    }

    public void setInfo(String info) {
        this.info = info;
    }

    public File getFile() {
        return file;
    }

    public void setFile(File file) {
        this.file = file;
    }

    /* CUSTOMIZE FUNCTIONS */

    @Override
    public BlockEle toBlockEle() {
        ImageBlockEle imageBlockEle = new ImageBlockEle();
        imageBlockEle.setInfo(info);
        imageBlockEle.setImageUrl(file.toUrl());

        return imageBlockEle;
    }
}

File: src/main/java/com/wishai/xzrtw/model/block/ImageBlock.kt
package com.wishai.xzrtw.model.block


import com.wishai.xzrtw.model.File
import com.wishai.xzrtw.model.gui.blocks.BlockEle
import com.wishai.xzrtw.model.gui.blocks.ImageBlockEle
import com.wishai.xzrtw.service.ResourceService
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.DependsOn
import org.springframework.stereotype.Component

import javax.persistence.*

@Entity
@Table(name = "block_image")
@DependsOn(value = ["XzrtPropertiesFactory"])
@ComponentScan("com.wishai.xzrtw.config")
class ImageBlock : Block() {

    var info: String? = null

    /* link to file */
    @OneToOne(cascade = arrayOf(CascadeType.ALL))
    @JoinColumn(name = "fileId")
    var file: File? = null

    init {
        super.setdType(ResourceService.BLOCK_TYPE_IMAGE)
    }

    /* CUSTOMIZE FUNCTIONS */

    override fun toBlockEle(): BlockEle {
        val imageBlockEle = ImageBlockEle()
        imageBlockEle.info = info
        imageBlockEle.imageUrl = file!!.toUrl()

        return imageBlockEle
    }
}
